File: backend/src/routes/analytics.ts
import { Router } from 'express'
import { prisma } from '../lib/prisma'
import { AnalyticsService } from '../services/AnalyticsService'

export function createAnalyticsRouter(service: Pick<AnalyticsService, 'getSummary'> = new AnalyticsService(prisma)): Router {
  const router = Router()
  router.get('/summary', async (_req, res, next) => {
    try { res.json({ success: true, data: await service.getSummary() }) }
    catch (error) { next(error) }
  })
  return router
}

export default createAnalyticsRouter()

File: backend/app/routes/analytics.py
"""运营分析路由。等价于原 TS routes/analytics.ts。"""
from __future__ import annotations

from fastapi import APIRouter

from ..deps import analytics_service
from ..errors import envelope

router = APIRouter()


@router.get("/summary")
async def summary():
    return envelope(analytics_service.get_summary())
